### PR TITLE
bump QEMU version to v2.9.1

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
xref #38067

**Special notes for your reviewer**:
/assign @luxas 

**Release note**:

```release-note
update QEMU version to v2.9.1
```

### CODE REVIEW

**Comment by dixudx** [comment]
/assign @jbeda @zmerlynn
I think it's ready for a review. PTAL. Thanks.

File: test/images/Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 REGISTRY ?= gcr.io/kubernetes-e2e-test-images
 GOARM=7
-QEMUVERSION=v2.7.0
+QEMUVERSION=v2.9.1
```
Comment (mkumatag) [inline test/images/Makefile]: Not really sure whether we need to push all the images under `test/images` with this new qemu binary. @luxas WDYT?

File: test/images/resource-consumer/BASEIMAGE [hunk 2]
```diff
@@ -1,4 +1,4 @@
-amd64=gcr.io/google-containers/debian-base-amd64:0.1
-arm=gcr.io/google-containers/debian-base-arm:0.1
-arm64=gcr.io/google-containers/debian-base-arm64:0.1
-ppc64le=gcr.io/google-containers/debian-base-ppc64le:0.1
+amd64=gcr.io/google-containers/debian-base-amd64:0.2
```
Comment (mkumatag) [inline test/images/resource-consumer/BASEIMAGE]: You may need to bump the version for resource-consumer image in `test/images/resource-consumer/VERSION`

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: build/root/WORKSPACE [hunk 3]
```diff
@@ -54,6 +54,7 @@ http_file(
     url = "https://storage.googleapis.com/kubernetes-release/network-plugins/cni-amd64-0799f5732f2a11b329d9e3d51b9c8f2e3759f2ff.tar.gz",
 )
 
+// TODO(dixudx): update the sha and tag (to v8) when the image get pushed
```
Comment (vabc3) [inline build/root/WORKSPACE]: Should use `#` for comments? Or spawn an issue for tracking

**Review by luxas — COMMENTED** [review]
Thanks @dixudx for your PR!

I don't think we need to bump the versions.
1) It shouldn't affect the result (this is only syscall emulation)
2) There is no direct benefit we get from using a newer QEMU that would need an image bump. This setting will just be picked up the next time the image is built/pushed.

The rationale for upgrading is to hopefully get more stable and reliable builds, have better emulation support for s390x and pick up performance improvements tha have been made in recent versions.

File: build/common.sh [hunk 4]
```diff
@@ -85,6 +85,7 @@ readonly KUBE_CONTAINER_RSYNC_PORT=8730
 #
 # $1 - server architecture
 kube::build::get_docker_wrapped_binaries() {
+  ## TODO(dixudx): update the tag to v8 when the image get pushed
```
Comment (luxas) [inline build/common.sh]: Is this still valid; do we need to do this?

**Review by luxas — COMMENTED** [review]
ping @ixdy; think we need to bump versions for this?

File: build/common.sh (hunk 4 above)
Comment (dixudx) [inline build/common.sh]: Since the base image of `debian-iptables`, we'd better bump a new version.

**Comment by ixdy** [comment]
QEMU is really only used when building the images, right? in that case, no, I don't think bumping the tags is needed.

**Comment by dixudx** [comment]
> QEMU is really only used when building the images, right?

Yes, we are just using it during building.

@ixdy Actually since the `Dockerfile` has been changed, will the docker images be updated and pushed to `gcr.io`? And If the images have been updated and pushed, but we still bump the old tag, users may not get noticed by the change. The image with the same tag they pulled before is actually different from that on current `gcr.io`.

**Comment by dixudx** [comment]
@ixdy @luxas @mkumatag 

> QEMU is really only used when building the images, right? in that case, no, I don't think bumping the tags is needed.

Updated. I've only bumped new version to QEMU. 

Since PR #50602 has to get bumped a new tag due to old stale base image. We just need to update it once.

**Review by mkumatag — APPROVED** [review]

**Comment by luxas** [comment]
@ixdy Please approve this. Thanks!

**Comment by dixudx** [comment]
ping @ixdy Need your approval. PTAL. Thanks!

**Pull request merged**